pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -239,6 +239,12 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 	// See if this has already been processed
 	if planExecution.Status.State == kudov1alpha1.PhaseStateComplete {
 		log.Printf("PlanExecutionController: PlanExecution \"%v\" has already run to completion, not processing.", planExecution.Name)
+		if instance.Status.ActivePlan.Name == planExecution.Name && instance.Status.ActivePlan.Namespace == planExecution.Namespace && instance.Status.Status != planExecution.Status.State {
+			// there was probably error because instance state is not in sync with PE state
+			instance.Status.Status = planExecution.Status.State
+			err = updateInstanceStatus(instance, planExecution.Status.State, r.Client)
```

Comment:
Updating the status of an instance from PlanExecutionController feels weird to me... do you think it would be better to queue a reconcile event for InstanceController?

We could use a [Channel Source](https://github.com/kubernetes-sigs/controller-runtime/blob/d7467fcf9e1c5817231f2768ffd626a244f2221b/pkg/source/example_test.go#L38-L50) for that.

Reply:
Another alternative which could be better is to add a Watch for PE events that enqueues a request for the instance that owns the PE.

Reply:
Oh yeah, you're right, that's probably better, thanks. I'll explore those options...

Reply:
I like the idea of using Channel Sources here. I think that could clean up a _lot_ of our code.

Reply:
Looks like watching channels is kinda discouraged and that a Watch for PE events triggering reconcile requests for InstanceController might be closer to the "Kubebuilder way":

https://book-v1.book.kubebuilder.io/beyond_basics/controller_watches.html

```
Watching Channels
Controllers may trigger Reconcile for events written to Channels. This is useful if the Controller needs to trigger a Reconcile in response to something other than a create / update / delete event to a Kubernetes object. Note: in most situations this case is better handled by updating a Kubernetes object with the external state that would trigger the Reconcile.
```
And https://godoc.org/sigs.k8s.io/controller-runtime/pkg/source#Channel:

```
Channel is used to provide a source of events originating outside the cluster (e.g. GitHub Webhook callback). Channel requires the user to wire the external source (eh.g. http handler) to write GenericEvents to the underlying channel.
```

Anyway I think that both approaches are better than having `PlanExecutionController` updating instances =).